pkg/cfn/builder/iam.go

```diff
@@ -431,6 +431,7 @@ func (rs *IAMServiceAccountResourceSet) AddAllResources() error {
 
 	if len(rs.spec.AttachPolicy) != 0 {
 		rs.template.NewResource("Policy1", &cft.IAMPolicy{
+			PolicyName:     cft.NewString(truncateResourceName(fmt.Sprintf("%s-%s", rs.spec.Namespace, rs.spec.Name))),
```

Comment:
My understanding was that this will get named automatically by CloudFormation, and setting a name will require `CAPABILITY_NAMED_IAM`. Could please clarify? Will CloudFormation pre-pend a unique suffix, but user is still required to provide the name?

Reply:
I can see that my understanding was wrong, we do set `PolicyName` in all other instances and it doesn't require `CAPABILITY_NAMED_IAM`.

Please see `pkg/cfn/template/iam_helpers.go`, we should just add `AttachPolicy` function there and use that. Be sure to use `MakeName` also, there is no need to have custom naming scheme here.